cropper/src/main/java/com/canhub/cropper/CropImage.java

```diff
@@ -275,17 +281,40 @@ public static Intent getCameraIntent(@NonNull Context context, Uri outputFileUri
         return allIntents;
     }
 
+
     /**
      * Get all Gallery intents for getting image from one of the apps of the device that handle
      * images.
      */
-    public static Intent getGalleryIntent(String action, boolean includeDocuments) {
-        Intent galleryIntent = new Intent();
-        galleryIntent.setAction(action);
-        galleryIntent.setType(includeDocuments ? "*/*" : "image/*");
-        galleryIntent.addCategory(Intent.CATEGORY_OPENABLE);
+    public static List<Intent> getGalleryIntents(
+            @NonNull PackageManager packageManager, String action, boolean includeDocuments) {
+        List<Intent> intents = new ArrayList<>();
+        Intent galleryIntent =
+                action == Intent.ACTION_GET_CONTENT
+                        ? new Intent(action)
+                        : new Intent(action, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        galleryIntent.setType("image/*");
+        List<ResolveInfo> listGallery = packageManager.queryIntentActivities(galleryIntent, 0);
+        for (ResolveInfo res : listGallery) {
+            Intent intent = new Intent(galleryIntent);
+            intent.setComponent(new ComponentName(res.activityInfo.packageName, res.activityInfo.name));
+            intent.setPackage(res.activityInfo.packageName);
+            intents.add(intent);
+        }
 
-        return galleryIntent;
+        // remove documents intent
+        if (!includeDocuments) {
+            for (Intent intent : intents) {
+                if (intent
+                        .getComponent()
+                        .getClassName()
+                        .equals("com.android.documentsui.DocumentsActivity")) {
+                    intents.remove(intent);
+                    break;
+                }
+            }
+        }
+        return intents;
```

Comment:
This code was changed because of the problem we have on latest OS 29 (when no legacy storage is enable) and 30. Android's 10 and 11 changed the storage access from external into scope storage. Bringing the old code back will bring a know issue back.

did you had the opportunity to test the library in the latests OS?

Reply:
oh. I see. I am using Pixel 4 with android 11 and have no problem with this code.

Didn't notice about #5. I will try to test more and let you know.